src/web/transactions.rs

```diff
@@ -1,122 +1,103 @@
-use std::collections::BTreeSet;
-
 use crate::{iroha_client_wrap::QueryBuilder, web::etc::HashDeser};
 
 use super::{
-    etc::{SerScaleHex, SignatureDTO, Timestamp},
+    etc::{SerScaleHex, Timestamp},
     get, web, AppData, Paginated, PaginationQueryParams, Scope, WebError,
 };
 use color_eyre::{eyre::Context, Result};
-use iroha_core::tx::{
-    Executable, RejectedTransaction, TransactionRejectionReason, TransactionValue, Txn,
-};
-use iroha_crypto::{Hash, HashOf, Signature};
+use iroha_core::tx::{Executable, TransactionValue, VersionedSignedTransaction};
+use iroha_crypto::{HashOf, SignaturesOf};
+use iroha_data_model::block::CommittedBlock;
 use iroha_data_model::prelude::{
-    FindAllTransactions, FindTransactionByHash, Instruction, Payload, SignedTransaction, UnlimitedMetadata,
+    FindAllTransactions, FindTransactionByHash, InstructionBox, TransactionQueryResult,
+    UnlimitedMetadata,
 };
+use iroha_data_model::transaction::{
+    error::model::TransactionRejectionReason, model::TransactionPayload,
+};
+
+use core::num::{NonZeroU32, NonZeroU64};
 use serde::Serialize;
 
 #[derive(Serialize)]
 #[serde(tag = "t", content = "c")]
 pub enum TransactionDTO {
-    Committed(CommittedTransactionDTO),
-    Rejected(RejectedTransactionDTO),
+    SingleTransactionDTO {
+        #[serde(flatten)]
+        base: TransactionBase,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        rejection_reason: Option<SerScaleHex<TransactionRejectionReason>>,
+    },
 }
```

Comment:
I think you can simply transform `TransactionDTO` to a structure:

```rust
struct TransactionDTO {
    #[serde(flatten)]
    base: TransactionBase,
    #[serde(skip_serializing_if = "Option::is_none")]
    rejection_reason: Option<SerScaleHex<TransactionRejectionReason>>
}
```

Reply:
Implemented in the latest commit as :
````
#[derive(Serialize)]
pub struct TransactionDTO {
    hash: SerScaleHex<HashOf<VersionedSignedTransaction>>,
    block_hash: SerScaleHex<HashOf<CommittedBlock>>,
    payload: TransactionPayloadDTO,
    signatures: BTreeSet<SignatureDTO>,
    #[serde(skip_serializing_if = "Option::is_none")]
    rejection_reason: Option<SerScaleHex<TransactionRejectionReason>>,
}
```